core/src/main/java/org/axonframework/config/EventProcessingConfigurer.java

```diff
@@ -0,0 +1,444 @@
+     */
+    default EventProcessingConfigurer registerTrackingEventProcessor(String name,
+                                                                     Function<Configuration, StreamableMessageSource<TrackedEventMessage<?>>> source) {
+        return registerTrackingEventProcessor(name,
+                                              source,
+                                              c -> c.getComponent(TrackingEventProcessorConfiguration.class,
+                                                                  TrackingEventProcessorConfiguration::forSingleThreadedProcessing));
+    }
+
+    /**
+     * Registers a {@link org.axonframework.eventhandling.TrackingEventProcessor} with given {@code name}, {@code
+     * source} and {@code processorConfiguration} within this Configurer.
+     *
+     * @param name                   The name of {@link org.axonframework.eventhandling.TrackingEventProcessor}
+     * @param source                 A function that builds Streamable Message Source
+     * @param processorConfiguration A function that builds Tracking Event Processor Configuration
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerTrackingEventProcessor(String name,
+                                                             Function<Configuration, StreamableMessageSource<TrackedEventMessage<?>>> source,
+                                                             Function<Configuration, TrackingEventProcessorConfiguration> processorConfiguration);
+
+    /**
+     * Registers a factory that builds the default Event Processor - Event Processor Builder to be used when there is no
+     * specific builder for given processor name.
+     *
+     * @param eventProcessorBuilder A function that builds Event Processor
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerEventProcessorFactory(EventProcessorBuilder eventProcessorBuilder);
+
+    /**
+     * Registers a Event Processor Builder for given processor {@code name}.
+     *
+     * @param name                  The name of the processor
+     * @param eventProcessorBuilder A function that builds Event Processor
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerEventProcessor(String name,
+                                                     EventProcessorBuilder eventProcessorBuilder);
+
+    /**
+     * Register a function that builds a Token Store for given {@code processingGroup}.
+     *
+     * @param processingGroup   The name of processing group
+     * @param tokenStoreBuilder A function that builds a Token Store
+     * @return an instance of Event Processing Configurer for fluent interfacing
+     */
+    EventProcessingConfigurer registerTokenStore(String processingGroup,
```

Comment:
A TokenStore is configured on the processor level (and only for tracking processors). Does the first parameter need to be named `eventProcessorName` ?

Reply:
You're right!